### ISSUE TITLE
Default queries and the ones in gitbase differ a lot

### ISSUE THREAD

**Issue by jfontan** [src-d/regression-gitbase#7]
The same query ID should always match the same query or one that has the same results. It's also a good time to renumber quer queries and take out the disabled ones. For example the ones that iterate all `files` table.

Default queries (https://github.com/src-d/regression-gitbase/blob/master/query.go#L14):

```go
var DefaultQueries = []Query{
	{
		ID:   "query0",
		Name: "All commits",
		Statements: []string{
			"SELECT * FROM commits",
		},
	},
	{
		ID:   "query1",
		Name: "Last commit messages in HEAD for every repository",
		Statements: []string{
			`SELECT c.commit_message FROM refs r JOIN commits c ON r.commit_hash = c.commit_hash WHERE r.ref_name = 'HEAD'`,
		},
	},
	{
		ID:   "query2",
		Name: "All commit messages in HEAD history for every repository",
		Statements: []string{
			`SELECT c.commit_message FROM commits c NATURAL JOIN ref_commits r WHERE r.ref_name = 'HEAD'`,
		},
	},
	{
		ID:   "query3",
		Name: "Top 10 repositories by commit count in HEAD",
		Statements: []string{
			`SELECT repository_id,commit_count FROM (SELECT r.repository_id,count(*) AS commit_count FROM ref_commits r WHERE r.ref_name = "HEAD" GROUP BY r.repository_id) AS q ORDER BY commit_count DESC LIMIT 10`,
		},
	},
	{
		ID:   "query4",
		Name: "Top 10 repositories by contributor count (all branches)",
		Statements: []string{
			`SELECT repository_id,contributor_count FROM (SELECT repository_id, COUNT(DISTINCT commit_author_email) AS contributor_count FROM commits GROUP BY repository_id) AS q ORDER BY contributor_count DESC LIMIT 10`,
		},
	},
	// {
	// 	ID:   "query5",
	// 	Name: "Create pilosa index on language UDF",
	// 	Statements: []string{
	// 		`CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)`,
	// 	},
	// },
	// {
	// 	ID:   "query7",
	// 	Name: "Query by language using the pilosa index",
	// 	Statements: []string{
	// 		`CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)`,
	// 		`SELECT file_path FROM files WHERE language(file_path, blob_content) = 'Go'`,
	// 		`DROP INDEX language_idx ON files`,
	// 	},
	// },
	// {
	// 	ID:   "query10",
	// 	Name: "Get all LICENSE blobs using pilosa index",
	// 	Statements: []string{
	// 		`CREATE INDEX file_path_idx ON files USING pilosa (file_path) WITH (async = false)`,
	// 		`SELECT blob_content FROM files WHERE file_path = 'LICENSE'`,
	// 	},
	// },
	{
		ID:   "query12",
		Name: "10 top repos by file count in HEAD",
		Statements: []string{
			`SELECT repository_id, num_files FROM (SELECT COUNT(f.*) num_files, f.repository_id FROM ref_commits r INNER JOIN commit_files cf ON r.commit_hash = cf.commit_hash AND r.repository_id = cf.repository_id INNER JOIN files f ON cf.repository_id = f.repository_id AND cf.blob_hash = f.blob_hash AND cf.tree_hash = f.tree_hash AND cf.file_path = f.file_path WHERE r.ref_name = 'HEAD' GROUP BY f.repository_id) t ORDER BY num_files DESC LIMIT 10`,
		},
	},
	{
		ID:   "query13",
		Name: "Top committers per repository",
		Statements: []string{
			`SELECT * FROM (SELECT commit_author_email as author,repository_id as id,count(*) as num_commits FROM commits GROUP BY commit_author_email, repository_id) t ORDER BY num_commits DESC`,
		},
	},
	{
		ID:   "query14",
		Name: "Top committers in all repositories",
		Statements: []string{
			`SELECT * FROM (SELECT commit_author_email as author,count(*) as num_commits FROM commits GROUP BY commit_author_email) t ORDER BY num_commits DESC`,
		},
	},
	// {
	// 	ID:   "query15",
	// 	Name: "Union operation with pilosa index",
	// 	Statements: []string{
	// 		`CREATE INDEX file_path_idx ON files USING pilosa (file_path) WITH (async = false)`,
	// 		`SELECT blob_content FROM files WHERE file_path = 'LICENSE' OR file_path = 'README.md'`,
	// 		`DROP INDEX file_path_idx ON files`,
	// 	},
	// },
	{
		ID:   "query17",
		Name: "Count all commits with NOT operation",
		Statements: []string{
			`SELECT COUNT(*) FROM commits WHERE NOT(commit_author_email = 'non existing email address')`,
		},
	},
}
```
Gitbase repo (https://github.com/src-d/gitbase/blob/c53388280b9d46e9dfa18c60e7e6231294cfabce/_testdata/regression.yml):

```yaml
---
-
  ID: 'query0'
  Name: 'All commits'
  Statements:
    - SELECT * FROM commits
-
  ID: 'query1'
  Name: 'Last commit messages in HEAD for every repository'
  Statements:
    - SELECT c.commit_message FROM refs r JOIN commits c ON r.commit_hash = c.commit_hash WHERE r.ref_name = 'HEAD'
-
  ID: 'query2'
  Name: 'All commit messages in HEAD history for every repository'
  Statements:
    - SELECT c.commit_message FROM commits c NATURAL JOIN ref_commits r WHERE r.ref_name = 'HEAD'
-
  ID: 'query3'
  Name: 'Top 10 repositories by commit count in HEAD'
  Statements:
    - SELECT repository_id,commit_count FROM (SELECT r.repository_id,count(*) AS commit_count FROM ref_commits r WHERE r.ref_name = 'HEAD' GROUP BY r.repository_id) AS q ORDER BY commit_count DESC LIMIT 10
-
  ID: 'query4'
  Name: 'Top 10 repositories by contributor count (all branches)'
  Statements:
    - SELECT repository_id,contributor_count FROM (SELECT repository_id, COUNT(DISTINCT commit_author_email) AS contributor_count FROM commits GROUP BY repository_id) AS q ORDER BY contributor_count DESC LIMIT 10
-
  ID: 'query6'
  Name: 'Create pilosa index on language UDF'
  Statements:
    - CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)
    - DROP INDEX language_idx ON files
-
  ID: 'query8'
  Name: 'Query by language using the pilosa index'
  Statements:
    - CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)
    - SELECT file_path FROM files WHERE language(file_path, blob_content) = 'Go'
    - DROP INDEX language_idx ON files
-
  ID: 'query9'
  Name: 'Query all files from HEAD'
  Statements:
    - SELECT cf.file_path, f.blob_content FROM ref_commits r NATURAL JOIN commit_files cf NATURAL JOIN files f WHERE r.ref_name = 'HEAD' AND r.history_index = 0
-
  ID: 'query11'
  Name: 'Get all LICENSE blobs using pilosa index'
  Statements:
    - CREATE INDEX file_path_idx ON files USING pilosa (file_path) WITH (async = false)
    - SELECT blob_content FROM files WHERE file_path = 'LICENSE'
    - DROP INDEX file_path_idx ON files
-
  ID: 'query12'
  Name: '10 top repos by file count in HEAD'
  Statements:
    - SELECT repository_id, num_files FROM (SELECT COUNT(f.*) num_files, f.repository_id FROM ref_commits r INNER JOIN commit_files cf ON r.commit_hash = cf.commit_hash AND r.repository_id = cf.repository_id INNER JOIN files f ON cf.repository_id = f.repository_id AND cf.blob_hash = f.blob_hash AND cf.tree_hash = f.tree_hash AND cf.file_path = f.file_path WHERE r.ref_name = 'HEAD' GROUP BY f.repository_id) t ORDER BY num_files DESC LIMIT 10
-
  ID: 'query13'
  Name: 'Top committers per repository'
  Statements:
    - SELECT * FROM (SELECT commit_author_email as author, repository_id as id, count(*) as num_commits FROM commits GROUP BY commit_author_email, repository_id) AS t ORDER BY num_commits DESC
-
  ID: 'query14'
  Name: 'Top committers in all repositories'
  Statements:
    - SELECT * FROM (SELECT commit_author_email as author,count(*) as num_commits FROM commits GROUP BY commit_author_email) t ORDER BY num_commits DESC
-
  ID: 'query16'
  Name: 'Union operation with pilosa index'
  Statements:
  - CREATE INDEX file_path_idx ON files USING pilosa (file_path) WITH (async = false)
  - SELECT blob_content FROM files WHERE file_path = 'LICENSE' OR file_path = 'README.md'
  - DROP INDEX file_path_idx ON files
-
  ID: 'query17'
  Name: 'Count all commits with NOT operation'
  Statements:
  - SELECT COUNT(*) FROM commits WHERE NOT(commit_author_email = 'non existing email address');
-
  ID: 'query18'
  Name: 'Count all commits with NOT operation and pilosa index'
  Statements:
  - CREATE INDEX email_idx ON commits USING pilosa (commit_author_email) WITH (async = false)
  - SELECT COUNT(*) FROM commits WHERE NOT(commit_author_email = 'non existing email address')
  - DROP INDEX email_idx ON commits
```

**Comment by kuba--** [comment]
@jfontan I think it's not a problem to sync up the numbers, but the idea is that either you run queries from gitbase or default ones from regression.

Moreover, after enumerating queries in gitbase and in regression, queries with the same numbers `queryXY` are exactly the same. 
If there is a _unique query_ it also should have a unique number.

**Comment by jfontan** [comment]
I see some ID's that don't match or queries are very similar:

```
	// {
	// 	ID:   "query5",
	// 	Name: "Create pilosa index on language UDF",
	// 	Statements: []string{
	// 		`CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)`,
	// 	},
	// },

VS

-
  ID: 'query6'
  Name: 'Create pilosa index on language UDF'
  Statements:
    - CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)
    - DROP INDEX language_idx ON files
```

```
	// {
	// 	ID:   "query7",
	// 	Name: "Query by language using the pilosa index",
	// 	Statements: []string{
	// 		`CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)`,
	// 		`SELECT file_path FROM files WHERE language(file_path, blob_content) = 'Go'`,
	// 		`DROP INDEX language_idx ON files`,
	// 	},
	// },

VS

-
  ID: 'query8'
  Name: 'Query by language using the pilosa index'
  Statements:
    - CREATE INDEX language_idx ON files USING pilosa (language(file_path, blob_content)) WITH (async = false)
    - SELECT file_path FROM files WHERE language(file_path, blob_content) = 'Go'
    - DROP INDEX language_idx ON files
```

```
	// {
	// 	ID:   "query10",
	// 	Name: "Get all LICENSE blobs using pilosa index",
	// 	Statements: []string{
	// 		`CREATE INDEX file_path_idx ON files USING pilosa (file_path) WITH (async = false)`,
	// 		`SELECT blob_content FROM files WHERE file_path = 'LICENSE'`,
	// 	},
	// },

VS

-
  ID: 'query11'
  Name: 'Get all LICENSE blobs using pilosa index'
  Statements:
    - CREATE INDEX file_path_idx ON files USING pilosa (file_path) WITH (async = false)
    - SELECT blob_content FROM files WHERE file_path = 'LICENSE'
    - DROP INDEX file_path_idx ON files
```

**Comment by kuba--** [comment]
Right, these are very similar, but as you see in `regression-gitbase` queries are commented (because are not supported by old gitbase.
So we can sync-up numbers but at the maybe we can remove commented queries, or maybe we can move all queries to one repo (instead of maintaining in both places)

**Comment by jfontan** [comment]
Right now only the default ones (from `regression-gitbase`) are run as there is an issue getting the yaml from the repositories, I'm working on fixing this.

I would leave only the ones uncommented in `regression-gitbase` as those are the ones being tested now. Unless there's some query that you feel should be checked. I would also renumber them. I wouldn't add any index query as there is no way to check performance for just querying it.

I feel it's better to keep the queries for regression in `gitbase` repository, same thing as tests. The ones in `regression-gitbase` should be a fallback when that file is not found.

What do you think @src-d/data-retrieval ?

**Comment by kuba--** [comment]
I wonder how many releases back, we wanna test?
Maybe all prev. gitbase releases already have `regression.yml`.
So if we remove one source it will simplify maintenance and ordering a lot. If not let's move it to some common repo and define the lowest supported version per query.

**Comment by jfontan** [comment]
I would not use it to test previous releases. For now it is going to test `master` automatically and in the future we can check  pull requests so I would have queries that work on `v0.18.0`. I believe that testing old versions is not a priority.

**Comment by kuba--** [comment]
So actually we can get rid of the default queries in regression repo. and just focus what's in yaml file. Right?

**Comment by jfontan** [comment]
Fixed with src-d/gitbase#666 and #9.
